### PR TITLE
[Docs] Add deep dive on a block's lifecycle

### PR DESCRIPTION
**Pull request by johnnyomair**
---

[Deploy preview](https://deploy-preview-2331.docs.comet-dxp.com/docs/blocks/lifecycle)

## PR Checklist

-   [x] Link to the respective task if one exists: [COM-766](https://vivid-planet.atlassian.net/browse/COM-766)

[COM-766]: https://vivid-planet.atlassian.net/browse/COM-766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs/blocks/lifecycle/lifecycle-admin.svg
Comment (johnnyomair) [inline docs/docs/blocks/lifecycle/lifecycle-admin.svg:1]: Two questions regarding the diagrams:

1. Should I add the original drawio files to the repo as well?
2. Should I invest some time into making them dark mode ready?

Comment (dkarnutsch) [reply docs/docs/blocks/lifecycle/lifecycle-admin.svg:1]: Maybe use https://docusaurus.io/docs/markdown-features/diagrams

Comment (johnnyomair) [reply docs/docs/blocks/lifecycle/lifecycle-admin.svg:1]: This is nice! 🤯

Comment (johnnyomair) [reply docs/docs/blocks/lifecycle/lifecycle-admin.svg:1]: Changed: ad65fdc17596c92c3d461b5ad4afab3c1c3d8ee0

The resulting diagrams are not that "beautiful" as the custom made ones, but they're way less effort.

Comment (dkarnutsch) [reply docs/docs/blocks/lifecycle/lifecycle-admin.svg:1]: The main benefits for me are:
- SSOT (no need for different files / providers)
- searchable
- easier renames

this outweighs the visual downsides, so I'm good with that 👍 

Do you think this requires a decision by the team?

**Review by dkarnutsch — APPROVED** [review]

File: docs/docs/blocks/lifecycle/lifecycle-admin.svg
Comment (johnnyomair) [reply docs/docs/blocks/lifecycle/lifecycle-admin.svg:1]: > Do you think this requires a decision by the team?

I don't think so. Also, it's team who's reviewing this PR 😁 

We could probably invest some effort into styling the diagrams, but that's of low priority to me.

**Review by nsams — APPROVED** [review]

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**